### PR TITLE
Added Graph Attention Network example

### PR DESCRIPTION
**Pull request by ebrahimpichka**
Hi,

Added Graph Attention Network example. Took the dataset processing procedure from the gcn example with some changes to be consistent with the existing gnn example.
would love to hear feedback.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -177,3 +177,14 @@ experiment with PyTorch.
     This example implements the `Semi-Supervised Classification with Graph Convolutional Networks <https://arxiv.org/pdf/1609.02907.pdf>`__ paper on the CORA database.
 
     `GO TO EXAMPLE <https://github.com/pytorch/examples/blob/main/gcn>`__ :opticon:`link-external` 
+
```
Comment (msaroufim) [inline docs/source/index.rst:180]: Doc build is failing, most likely culprit is changes to this file

Comment (ebrahimpichka) [reply docs/source/index.rst:180]: Thanks for the review. I rechecked that.
My local build is served without any issues with sphinx in venv.
i'll keep looking to see where could the issue be

**Comment by ebrahimpichka** [comment]
First, it seemed, the issue was with the new example card in the doc.
After checking a few times, the build with the default commits (without my changes) once passed [ac8d3ef](https://github.com/pytorch/examples/pull/1174/commits/ac8d3efeaf3b4ef7116a8c13f0fbe95d62f4f7bb) and once failed 
[d5fe1d2](https://github.com/pytorch/examples/pull/1174/commits/d5fe1d2cc3d5851981eb278e6f82895a0547bd98) the checks.

Couldn't find the issue with the build since it builds locally with no issues.

**Comment by msaroufim** [comment]
@sekyondaMeta @svekars any idea what's going on with the doc build here?

**Comment by sekyondaMeta** [comment]
> @sekyondaMeta @svekars any idea what's going on with the doc build here?

I wonder if this failure here is the culprit: 
https://github.com/pytorch/examples/actions/runs/5627940947/job/15251219562?pr=1174

One of the changes added numpy 1.25.1 and there is probably a requirement somewhere else for 1.24.2

**Comment by ebrahimpichka** [comment]
@sekyondaMeta thanks, removed the version specification from the requirements file

**Comment by msaroufim** [comment]
@svekars could I please get access to the netlify logs? I can't see why this is failing

**Comment by ebrahimpichka** [comment]
Thanks @msaroufim 
I did not add the doc card for the example.
should I add it in another PR or it's not required?

**Comment by msaroufim** [comment]
Make a separate PR. I'm still not sure what's going on with docbuikd since I can't read the build failures

**Pull request merged**